java/piranha/src/test/java/com/uber/piranha/EnumConstantTest.java

```diff
@@ -225,4 +683,43 @@ public void testRemoveEnumFieldValueMatchTest() {
             "}")
         .doTest();
   }
+
+  /**
+   * Test to cleanup an enum with only one remaining enum constant to be cleaned up. We expect this
+   * to fail and throw an exception.
+   *
+   * <p>TODO: Remove the entire enum if we remove the last constant from it This is invalid Java
+   * code
+   */
+  @Test(expected = IllegalStateException.class)
```

Comment:
What is the full exception? 

Interestingly, we should handle this for the case of treatment group enums, where we, either:
a) simply delete the enum if nested, _or_,
b) in the common case of a top level enum, leave it empty and add a comment that our postprocessing uses to remove the file
See [here](https://github.com/uber/piranha/blob/master/java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java#L732).

Wonder if there is any reason we couldn't do the same for flag/experiment enums... maybe instead of processing the enum constant to be deleted, plus the second to last constant, we could match on the full enum and then do all our checks and rewritings there (wonder if that's easier or harder, though).

Reply:
Hmm OK, so playing around with this, the issue is that it is a compile failure if we remove all constants AND the semicolon before the fields/methods/etc., BUT we can fix this and have valid compilation by putting the semicolon back in (or never removing it in the first place) despite there being no constants left.

So I can change this to make sure we never remove the very last semi-colon and then we will no longer get compilation failure.